File: Python/Pakaian.py
# Class turunan Produk: data pakaian
# Atribut: ukuran, bahan, warna
from Produk import Produk


class Pakaian(Produk):
    # Memanggil constructor milik Produk lewat super()
    def __init__(self, id_produk="", merk="", harga=0.0,
                 ukuran="", bahan="", warna=""):
        super().__init__(id_produk, merk, harga)
        self.ukuran = ukuran
        self.bahan = bahan
        self.warna = warna

    # Setter dan getter
    def set_ukuran(self, ukuran):
        self.ukuran = ukuran

    def get_ukuran(self):
        return self.ukuran

    def set_bahan(self, bahan):
        self.bahan = bahan

    def get_bahan(self):
        return self.bahan

    def set_warna(self, warna):
        self.warna = warna

    def get_warna(self):
        return self.warna

    # Destructor
    def __del__(self):
        pass

File: CPP/Pakaian.cpp
#include "Produk.cpp"

// Class turunan Produk: data pakaian
// Atribut: ukuran, bahan, warna
class Pakaian : public Produk{
    protected:
        string ukuran;
        string bahan;
        string warna;

    public:
        // Constructor
        Pakaian() {}

        Pakaian(string id_produk, string merk, float harga, string ukuran, string bahan, string warna) : Produk(id_produk, merk, harga) {
            this->ukuran = ukuran;
            this->bahan = bahan;
            this->warna = warna;
        }

        // Setter dan getter for ukuran
        void setUkuran(string ukuran) {this->ukuran = ukuran;}
        string getUkuran() {return ukuran;}

        // Setter dan getter for bahan
        void setBahan(string bahan) {this->bahan = bahan;}
        string getBahan() {return bahan;}

        // Setter dan getter for warna
        void setWarna(string warna) {this->warna = warna;}
        string getWarna() {return warna;}

        // Destructor
        ~Pakaian() {}
};
